pkg/controller/v1alpha1/trainedmodel/controller.go

```diff
@@ -190,6 +199,45 @@ func (r *TrainedModelReconciler) updateStatus(req ctrl.Request, desiredModel *v1
 	return nil
 }
 
+func (r *TrainedModelReconciler) updateConditions(req ctrl.Request, tm *v1alpha1api.TrainedModel) error {
+	// Get the parent inference service
+	isvc := &v1beta1api.InferenceService{}
+	if err := r.Get(context.TODO(), types.NamespacedName{Namespace: req.Namespace, Name: tm.Spec.InferenceService}, isvc); err != nil {
+		return err
+	}
+
+	var conditionErr error = nil
+	// Update Inference Service Ready condition
+	if isvc.Status.IsReady() {
+		log.Info("Parent InferenceService is ready", "TrainedModel", tm.Name, "InferenceService", isvc.Name)
+		tm.Status.SetCondition(v1alpha1api.InferenceServiceReady, &apis.Condition{
+			Status: v1.ConditionTrue,
+		})
+	} else {
+		log.Info("Parent InferenceService is not ready", "TrainedModel", tm.Name, "InferenceService", isvc.Name)
+		tm.Status.SetCondition(v1alpha1api.InferenceServiceReady, &apis.Condition{
+			Type:    v1alpha1api.InferenceServiceReady,
+			Status:  v1.ConditionFalse,
+			Reason:  "InferenceServiceNotReady",
+			Message: "Inference Service needs to be ready before Trained Model can be ready",
+		})
+
+		conditionErr = utils.FirstNonNilError([]error{
+			conditionErr,
+			fmt.Errorf(InferenceServiceNotReady, isvc.Name, tm.Name),
+		})
+	}
+
+	if conditionErr != nil {
+		if statusErr := r.Status().Update(context.TODO(), tm); statusErr != nil {
+			r.Recorder.Eventf(tm, v1.EventTypeWarning, "UpdateFailed",
+				"Failed to update status for TrainedModel %q: %v", tm.Name, conditionErr)
+		}
+	}
+
+	return conditionErr
```

Comment:
feels a bit not intuitive to treat condition not ready the same way as a generic Get/Update error, because the caller just ignores the error and return without requeue. ideally for Get/Update error should return err and requeue immediately.

Reply:
That's another mistake I made. It shouldn't have ignored the returned error. Changed it to do so. Just to clarify, returning an error will make the it requeue?

Reply:
yes, there will be requeue with exponential backoff when reconciler return error.